drivers/kscan/kscan_cap1203.c

```diff
@@ -0,0 +1,267 @@
+{
+	struct cap1203_data *data = CONTAINER_OF(timer, struct cap1203_data, timer);
+
+	k_work_submit(&data->work);
+}
+#endif
+
+static int cap1203_configure(const struct device *dev,
+			     kscan_callback_t callback)
+{
+	struct cap1203_data *data = dev->data;
+	const struct cap1203_config *config = dev->config;
+	int r;
+
+	if (!callback) {
+		LOG_ERR("Invalid callback (NULL)");
+		return -EINVAL;
+	}
+
+	data->callback = callback;
+
+	if (config->int_gpio.port != NULL) {
+		/* Clear pending interrupt */
+		r = cap1203_clear_interrupt(&config->i2c);
+		if (r < 0) {
+			return r;
+		}
+
+		r = cap1203_enable_interrupt(&config->i2c, INTERRUPT_ENABLE);
+	} else {
+		r = cap1203_enable_interrupt(&config->i2c, INTERRUPT_DISABLE);
+	}
+
+	if (r < 0) {
+		LOG_ERR("Could not configure interrupt");
+		return r;
+	}
+
+	return 0;
+}
+
+static int cap1203_enable_callback(const struct device *dev)
+{
+	struct cap1203_data *data = dev->data;
+
+	const struct cap1203_config *config = dev->config;
+
+	if (config->int_gpio.port != NULL) {
+		const struct cap1203_config *config = dev->config;
+
+		gpio_add_callback(config->int_gpio.port, &data->int_gpio_cb);
+	}
+#ifdef CONFIG_KSCAN_CAP1203_POLL
+	else {
+		k_timer_start(&data->timer, K_MSEC(CONFIG_KSCAN_CAP1203_PERIOD),
+			      K_MSEC(CONFIG_KSCAN_CAP1203_PERIOD));
+	}
+#endif
+	return 0;
+}
+
+static int cap1203_disable_callback(const struct device *dev)
+{
+	struct cap1203_data *data = dev->data;
+
+	const struct cap1203_config *config = dev->config;
+
+	if (config->int_gpio.port != NULL) {
+		const struct cap1203_config *config = dev->config;
+
+		gpio_remove_callback(config->int_gpio.port, &data->int_gpio_cb);
+	}
+#ifdef CONFIG_KSCAN_CAP1203_POLL
+	else {
+		k_timer_stop(&data->timer);
+	}
+#endif
+	return 0;
+}
+
+static int cap1203_init(const struct device *dev)
+{
+	const struct cap1203_config *config = dev->config;
+	struct cap1203_data *data = dev->data;
+	int r = -ENOTSUP;
+
+	if (!device_is_ready(config->i2c.bus)) {
+		LOG_ERR("I2C controller device not ready");
+		return -ENODEV;
+	}
+
+	data->dev = dev;
+
+	k_work_init(&data->work, cap1203_work_handler);
+
+	if (config->int_gpio.port != NULL) {
+		if (!device_is_ready(config->int_gpio.port)) {
+			LOG_ERR("Interrupt GPIO controller device not ready");
+			return -ENODEV;
+		}
+
+		r = gpio_pin_configure_dt(&config->int_gpio, GPIO_INPUT);
+		if (r < 0) {
+			LOG_ERR("Could not confighure interrupt GPIO pin");
+			return r;
+		}
+
+		r = gpio_pin_interrupt_configure_dt(&config->int_gpio,
+						   GPIO_INT_EDGE_TO_ACTIVE);
+		if (r < 0) {
+			LOG_ERR("Could not configure interrupt GPIO interrupt");
+			return r;
+		}
+
+		gpio_init_callback(&data->int_gpio_cb, cap1203_isr_handler,
+				   BIT(config->int_gpio.pin));
+	}
+#ifdef CONFIG_KSCAN_CAP1203_POLL
+	else {
+		k_timer_init(&data->timer, cap1203_timer_handler, NULL);
+		r = 0;
+	}
+#endif
+
+	if (r < 0) {
+		LOG_ERR("Could not configure device");
+	}
+
+	return r;
```

Comment:
this path seems unfeasible, as you always log error and return directly, just`return 0` at the end.

```suggestion
	return 0;
```

Reply:
OK, I'll fix it.